libs/waku-utils/waku_handshake_utils.nim

```diff
@@ -165,46 +142,82 @@ proc prepareHandShakeMsg*(rng: ref HmacDrbgContext,
   assert wakuMsg.get().contentTopic == contentTopic
   agentMessageNametag = toMessageNametag(agentHS)
   wakuMsg
 
-proc publishHandShakeMsg*(node: WakuNode,
-                          pubSubTopic: PubsubTopic,
-                          contentTopic: ContentTopic,
-                          message: WakuMessage,
-                          step: int) {.async.} =
+proc publishHandShakeMsg*(node: WakuNode, pubSubTopic: PubsubTopic,
+    contentTopic: ContentTopic, message: WakuMessage, step: int) {.async.} =
   notice "Publishing handshake message for step:", step = step
-  await sleepAsync(5000)
   await node.publish(some(pubSubTopic), message)
   notice "Published handshake message for step:", step = step,
-         psTopic = pubSubTopic,
-         contentTopic = contentTopic,
-         message = message
-
-proc handleHandShakeMsg*(rng: ref HmacDrbgContext,
-                         pubSubTopic: PubsubTopic,
-                         contentTopic: ContentTopic,
-                         step: int,
-                         payload: PayloadV2,
-                         initiatorStep: var HandshakeStepResult,
-                         initiatorHS: var HandshakeState,
-                         initiatorMessageNametag: var MessageNametag) =
+      psTopic = pubSubTopic, contentTopic = contentTopic, message = message
+
+proc handleHandShakeMsg*(rng: ref HmacDrbgContext, pubSubTopic: PubsubTopic,
+    contentTopic: ContentTopic, step: int, payload: PayloadV2,
+    initiatorStep: var HandshakeStepResult, initiatorHS: var HandshakeState,
+    initiatorMessageNametag: var MessageNametag) =
   notice "Received handshake message for step:", step = step,
          psTopic = pubSubTopic,
          contentTopic = contentTopic,
          payload = payload
   notice "Handling handshake message for step:", step = step
-  initiatorStep = stepHandshake(rng[], initiatorHS,
-                               readPayloadV2 = payload,
-                               messageNametag = initiatorMessageNametag).get()
+  initiatorStep = stepHandshake(rng[], initiatorHS, readPayloadV2 = payload,
+      messageNametag = initiatorMessageNametag).get()
   initiatorMessageNametag = toMessageNametag(initiatorHS)
 
+proc initiatorHandshake*(rng: ref HmacDrbgContext, node: WakuNode,
+    pubSubTopic: PubsubTopic, contentTopic: ContentTopic, qr: string,
+    qrMessageNameTag: seq[byte], initiatorInfo: AgentKeysAndCommitment
+   ): Future[HandshakeResult] {.async.} =
+  var
+    readyForFinalization = false
+
+    initiatorMessageNametag: MessageNametag
+    initiatorStep: HandshakeStepResult
+    initiatorHS: HandshakeState = initHS(initiatorInfo, qr, true)
+
+    initiatorHSResult: HandshakeResult
+
+  let initialMessage =
+    prepareHandShakeInitiatorMsg(rng, contentTopic, initiatorInfo,
+        qrMessageNameTag, initiatorMessageNametag, initiatorHS, initiatorStep)
+
+  proc initiatorHandshakeHandler(topic: PubsubTopic, msg: WakuMessage): Future[
+      void] {.async, gcsafe.} =
```

Comment:
Simple nitpick: the `async` pragma statement is equivalent to `gcsafe, raises: []`
```suggestion
      void] {.async.} =
```